components/nexus-repository/src/main/java/org/sonatype/nexus/repository/manager/RepositoryManagerImpl.java

```diff
@@ -217,6 +217,7 @@ protected void doStop() throws Exception {
     for (Repository repository : repositories.values()) {
       log.debug("Stopping repository: {}", repository);
       repository.stop();
+      store.update(repository.getConfiguration());
```

Comment:
this seems like a nightmare for future auditing: "Why did my configuration change?"